stip/stip-001-burn-election.md

```diff
@@ -0,0 +1,127 @@
+
+1. Near instantaneous confirmation of Stacks transactions
+2. High validation throughput
+3. An open leadership set
+4. Opportunity to participate without mining hardware
+5. Ability to migrate to a separate burn chain in the future
+
+# Election
+
+The basic structure for leader election through burn-mining is that
+for some Stacks block _N_, the leader is selected via some function of
+that leader's total cryptocurrency burnt in a block _N'_ on the
+underlying burn chain (or over a window of blocks).
+
+In such a system, if a candidate _Alice_ wishes to be a leader of a
+Stacks block, she issues a burn transaction in the underlying burn
+chain. The network then uses cryptographic sortition to choose a
+leader in a verifiably random process, weighted by burn amounts.
+The block in which this burn transaction is broadcasted is known
+as the "election block" for Stacks block _N_.
+
+## Committing to a chain tip
+
+Leaders must be able to choose which chain tip they want to build off
+of. Otherwise, malicious leaders would be able to force subsequent
+leaders to follow their chains, even if they misbehaved by hiding
+transactions, broadcasting partial blocks, etc.
+
+However, a healthy blockchain requires that leaders _commit_ to the
+chain tip they are working on at the time at which they expend energy,
+burn cryptocurrency, etc. This ensures that if a would-be malicious
+leader fails to win a block, they have suffered a penalty. For
+example, in Bitcoin, miners search for the inverse of a hash that
+contains the current chain tip. If they fail to win that block, they
+have wasted that energy. As they must attempt deeper and deeper forks
+to initiate a double-spend attacks, this becomes an exponential energy
+expenditure.
+
+It is important that this same property hold in the Stacks
+blockchain. The ability for leaders to engage in double-spend attacks
+at will is unacceptable. This property is enforced by making a leader
+announce their chain commitment _in the burn transaction_. However,
+unlike Bitcoin (and, for example, Bitcoin-NG) --- this commitment is a
+"lagging commitment". The chain tip announced in a burn transaction
+_cannot_ be the exact preceding chain tip, because a key advantage of
+using leader election is that a block may be _streamed_ -- namely, a
+leader is continuously confirming new transactions as long as their
+tenure lasts. So at a minimum, the commitment will be 1-block old.
+
+This time between block _N'_ on the burn chain and the resulting
+leader's tenure results in a trade-off between (1) resilience towards
+reorgs in the underlying blockchain and (2) period of time during
+which a malicious leader may knowingly engage in successful
+double-spend attacks. It is important to note that this trade-off
+exists regardless of fork selection algorithm--- any fork selection
+strategy will allow for a leader to know _a priori_ whether a
+malicious fork is likely to be successful.
```

Comment:
Hmm.  What if we made it so that the act of paying off the fork penalties didn't take effect until after an amount of time equal to the time between block _N'_ and the tenure begins passes?  For example, if the burn window is 144 blocks behind the chain tip, can we make it so the result of paying all outstanding fork penalties doesn't take effect until 288 blocks later?

Reply:
We need to think about the remediation techniques here a little more carefully -- I'm still unsure about how a delayed-forking scheme would work.